opencti-platform/opencti-front/src/private/components/riskAssessment/risks/remediations/RemediationDetailsPopover.js

```diff
@@ -102,23 +117,51 @@ class RemediationDetailsPopover extends Component {
     this.setState({ anchorEl: null });
   }
 
-  handleCloseUpdate() {
-    this.setState({ details: false });
+  handleSubmit() {
+    this.setState({ onSumbit: true });
   }
 
-  handleOpenDetails() {
-    this.setState({ details: true });
+  onReset() {
     this.handleClose();
   }
 
-  handleCloseDetails() {
-    this.setState({ details: false });
-  }
-
   handleCancelCloseClick() {
     this.setState({ close: false });
   }
 
+  onSubmit(values, { setSubmitting, resetForm }) {
+    const adaptedValues = R.evolve(
+      {
+        modified: () => values.modified === null ? null : parse(values.modified).format(),
+        created: () => values.created === null ? null : parse(values.created).format(),
+      },
+      values,
+    );
+    const finalValues = R.pipe(
+      R.toPairs,
+      R.map((n) => ({
+        'key': n[0],
+        'value': adaptFieldValue(n[1]),
+      })),
+    )(adaptedValues);
+    CM(environmentDarkLight, {
+      mutation: remediationEditionMutation,
+      variables: {
+        id: this.props.cyioCoreRelationshipId,
+        input: finalValues,
+      },
+      setSubmitting,
+      onCompleted: (data) => {
+        setSubmitting(false);
+        resetForm();
+        this.handleClose();
+        this.props.history.push(`/activities/risk assessment/risks/${this.props.riskId}/remediation/`);
+      },
+      onError: (err) => console.log('editRemediationMutationError', err),
```

Comment:
@sachin755 do not use `console.log` for errors. Use `console.error` and expose an error message to the user using `utils/bakesToast.js`.